src/ringmesh/io/geomodel_adapter_resqml.cpp

```diff
@@ -0,0 +1,677 @@
+
+                std::unique_ptr< double[] > points(
+                    new double[surface.nb_vertices() * 3] );
+
+                vec3 p;
+                for( auto v : range( surface.nb_vertices() ) )
+                {
+                    p = surface.vertex( v );
+                    points[v * 3] = p[0];
+                    points[v * 3 + 1] = p[1];
+                    points[v * 3 + 2] = p[2];
+                }
+
+                std::unique_ptr< unsigned int[] > node_indices(
+                    new unsigned int[surface.nb_mesh_elements() * 3] );
+                for( auto t : range( surface.nb_mesh_elements() ) )
+                {
+                    for( auto v :
+                        range( surface.nb_mesh_element_vertices( t ) ) )
+                    {
+                        node_indices[t * 3 + v] =
+                            interface_vertex_count
+                            + surface.mesh_element_vertex_index( { t, v } );
+                    }
+                }
+
+                rep->pushBackTrianglePatch( surface.nb_vertices(), &points[0],
+                    surface.nb_mesh_elements(), &node_indices[0], hdf_proxy_ );
+
+                interface_vertex_count += surface.nb_vertices();
+            }
+
+            if( interface_vertex_count == 0 )
+            {
+                return true;
+            }
+
+            std::vector< AbstractValuesProperty* > vertex_properties;
+            std::vector< AbstractValuesProperty* > polygon_properties;
+            for( auto i : range( interface.nb_children() ) )
+            {
+                const Surface3D& surface =
+                    static_cast< const Surface3D& >( interface.child( i ) );
+
+                write_property( i == 0, vertex_properties,
+                    surface.vertex_attribute_manager(),
+                    gsoap_resqml2_0_1::resqml2__IndexableElements__nodes,
+                    *rep );
+                write_property( i == 0, polygon_properties,
+                    surface.polygon_attribute_manager(),
+                    gsoap_resqml2_0_1::resqml2__IndexableElements__cells,
+                    *rep );
+            }
+        }
+        return true;
+    }
+
+    bool GeoModelAdapterRESQMLImpl::write_volumes()
+    {
+        for( const auto& layer :
+            geomodel_.geol_entities( Layer3D::type_name_static() ) )
+        {
+            AbstractFeatureInterpretation* interp = nullptr;
+            if( layer.has_geological_feature() )
+            {
+                AbstractFeature* feature = find_or_create_feature( layer );
+                AbstractFeatureInterpretation* interp =
+                    find_or_create_interpretation( *feature );
+            }
+            else
+            {
+                Logger::warn( "", layer.gmge(),
+                    " geological feature is empty or not recognized" );
+            }
+
+            // TODO: I do not understand why the interp is not needed...
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1135)